libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/DecCBOR.hs

```diff
@@ -107,7 +108,11 @@ class Typeable a => DecCBOR a where
   label = T.pack . show . typeRep
 
 instance DecCBOR Version where
-  decCBOR = decodeVersion
+  decCBOR =
+    ifDecoderVersionAtLeast
+      (natVersion @12)
+      decodeVersionW32
+      decodeVersion
```

Comment:
Sorry, but reviewing your PR made me realize that this change isn't really necessary, since it is not possible to construct `Version` with value higher than `MaxVersion` anyways: 
https://github.com/IntersectMBO/cardano-ledger/blob/468c9da4cf0cab1348e4fc4175abf3247d24f288/libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Version.hs#L54

Which is enforced by `mkVersion64`
But, it won't hurt either.
```suggestion
  decCBOR = decodeVersion
```